app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee

```diff
@@ -0,0 +1,76 @@
+class MS.Views.TimelineSelectStepsView extends Backbone.View
+
+  el: 'body'
+
+  events: (params) ->
+    'change .select-step': 'remember_checked'
+    'click #choose-all': 'control_all'
+    'click #check-all-ctrl': 'check_all'
+    'click #uncheck-all-ctrl': 'uncheck_all'
+    'click #check-completed-ctrl': 'check_completed'
+    'click #check-incomplete-ctrl': 'check_incomplete'
+ 
+  check_all: (e) ->
+    control = e.currentTarget
+    $inputs = $(control).closest(".timeline").find('.select-step')
+    this.check_items($inputs)
+
+    $master_checkbox = $(control).closest(".btn-group").find("#choose-all")
```

Comment:
When selecting something by ID you don't need to scope it like this. There should only be one of them on the whole page, so @$('#choose-all') should be sufficient. If there are more than one, should change to class instead of id.